builtin/logical/pki/storage.go

```diff
@@ -1394,6 +1394,22 @@ func (sc *storageContext) getAutoTidyConfig() (*tidyConfig, error) {
 		result.IssuerSafetyBuffer = defaultTidyConfig.IssuerSafetyBuffer
 	}
 
+	// These cases pass as default
+	// Case 1: SafetyBuffer && RevokedSafetyBuffer are default, RevokedSafetyBuffer is defaultTidyConfig.SafetyBuffer
+	// Case 2: SafetyBuffer is default && RevokedSafetyBuffer has been explicitly set: RevokedSafetyBuffer should be set value
+	// Case 3: SafetyBuffer && RevokedSafetyBuffer have been explicitly set: RevokedSafetyBuffer should be its set value
+
+	// FAILS
+	// Case 4: SafetyBuffer explicitly set && RevokedSafetyBuffer is unset: RevokedSafetyBuffer should be updated SafetyBuffer
```

Comment:
@cipherboy, this is my dilemma with case 4: Only safety_buffer is explicitly set.

I used printed the memory addresses for debugging, these are the results for the lines 1335 - 1344 of the test **TestRevokedSafetyBufferConfig** : 

    fmt.Printf("Address defaultTidyConfig RevokedSafetyBuffer is pointing to: %p\n", defaultTidyConfig.RevokedSafetyBuffer)
    fmt.Printf("Address of Default SafetyBuffer: %p\n", &defaultTidyConfig.SafetyBuffer)
    fmt.Printf("Address RevokedSafetyBuffer is pointing to: %p\n", result.RevokedSafetyBuffer)
    fmt.Printf("Value of *RevokedSafetyBuffer: %v\n", *result.RevokedSafetyBuffer)
    fmt.Printf("Address of RevokedSafetyBuffer: %v\n", &result.RevokedSafetyBuffer)
    fmt.Printf("Value of Default SafetyBuffer: %v\n", defaultTidyConfig.SafetyBuffer)
    fmt.Printf("Address of result.SafetyBuffer: %p\n", &result.SafetyBuffer)
    fmt.Printf("Value of result.SafetyBuffer: %v\n", result.SafetyBuffer)

```
Address defaultTidyConfig RevokedSafetyBuffer is pointing to: 0x106ce75f8
Address of Default SafetyBuffer: 0x106ce75f8
Address RevokedSafetyBuffer is pointing to: 0x1400031f8a0
Value of *RevokedSafetyBuffer: 72h0m0s
Address of RevokedSafetyBuffer: 0x140009fc160
Value of Default SafetyBuffer: 72h0m0s
Address of result.SafetyBuffer: 0x140009fc158
Value of result.SafetyBuffer: 1h0m0s
```

As seen above,` result.RevokedSafetyBuffer` does not point to `defaultTidyConfig.SafetyBuffer` if it is unset because it is reassigned here: 
https://github.com/openbao/openbao/blob/981935e68736b3da290442543b832871c77a421a/builtin/logical/pki/storage.go#L1383-L1387

Reply:
Hmmm, I think the clever use of pointers might be causing problems. Typically I think we've left it `nil` until a value has been set explicitly by the operator. This is a bit more complex as I think you'd need more fallback logic (in places, to ensure you read `safety_buffer` when `revoked_safety_buffer` is nil), but what do you think about that approach?

Reply:
Thanks you, this approach was less confusing to work with :)